### ISSUE TITLE
Add a similar cut out feature for screenshots just like the one in Screenpresto

### ISSUE THREAD

**Issue by AkechiShiro** [flameshot-org/flameshot#3524]
### Feature Description

I believe this feature would be a very good addition to be on feature par vs this proprietary tool that is available on Windows : 
<img width="749" alt="image" src="https://github.com/flameshot-org/flameshot/assets/14914796/89475b79-6ca3-4b6a-9288-9ea729659022">
The feature allows to cut an image/crop but either horizontally or vertically of the image

I have no idea how would one implement such a feature nor do I have read the codebase of flameshot or is familiar with it.

But I believe this feature would be a very good addition to Flameshot.

**Comment by borgmanJeremy** [comment]
@mmahmoudian  I don't think this is feasible with flameshot because of the way we launch in full screen. If you "cut" a section either vertically or horizontally then the rest of the image would need to stretch (or not be fully screen).

**Comment by mmahmoudian** [comment]
@borgmanJeremy i feel like this is something that can be done by exposing the coordinates of the selected region to a plugin. This operation doesn't sound like a complex operation. I think it can be done for example via the montage subcommand of imagemagic:

https://imagemagick.org/script/montage.php

I assume the same functionality is also available through their library and can be integrated in plugins.

But apart from this issue, and a bit off-topic, since it has come up in other issues as well, maybe we can also have a windowed mode. This can help address other issue such as:
1. Editing pinned images
2. Loading other images from disk and annotate them

**Comment by peitch** [comment]
@AkechiShiro I agree that this would be an excellent feature. Snagit  also has this feature in their editor. I think it is my favorite editing tool.

[https://www.techsmith.com/snagit/features/?type=editing](url)

![Image](https://github.com/user-attachments/assets/10b2d841-ab00-424f-965b-c2240bb34533)
